earn/src/pages/BoostPage.tsx

```diff
@@ -57,88 +84,131 @@ export default function BoostPage() {
-        const color1 = (await getProminentColor(pos.token1.logoURI)).replace(' ', '');
-        return [idx, [`rgb(${color0})`, `rgb(${color1})`]] as [number, [string, string]];
-      });
-
-      setColors(new Map(await Promise.all(entries)));
-    };
-
-    fetch();
-    return () => {};
-  }, [nonZeroUniswapNFTPositions]);
-
-  const contracts = useMemo(() => {
-    return nonZeroUniswapNFTPositions.map((position) => {
+  /*//////////////////////////////////////////////////////////////
+            FETCH UNISWAP NFT POSITIONS - CONT. (SLOT0)
+  //////////////////////////////////////////////////////////////*/
+  const poolContracts = useMemo(() => {
+    return uniswapNFTPositions.map((position) => {
       return {
         abi: UniswapV3PoolABI,
         address: computePoolAddress(position),
         functionName: 'slot0',
         chainId: activeChain.id,
       } as const;
     });
-  }, [activeChain.id, nonZeroUniswapNFTPositions]);
-
+  }, [activeChain.id, uniswapNFTPositions]);
   const { data: slot0Data } = useContractReads({
-    contracts: contracts,
+    contracts: poolContracts,
     allowFailure: false,
   });
 
-  const uniswapNFTCardInfo: BoostCardInfo[] = useMemo(() => {
-    if (slot0Data === undefined || slot0Data.length !== nonZeroUniswapNFTPositions.length) {
+  /*//////////////////////////////////////////////////////////////
+                     CREATE UNISWAP CARD INFOS
+  //////////////////////////////////////////////////////////////*/
+  const uniswapCardInfos: BoostCardInfo[] = useMemo(() => {
+    if (slot0Data === undefined || slot0Data.length !== uniswapNFTPositions.length) {
       return [];
     }
-    return nonZeroUniswapNFTPositions.map((position, index) => {
-      const currentTick = slot0Data[index][1];
+    return uniswapNFTPositions.map((position, index) => {
+      const currentTick = slot0Data[index]['tick'];
 
       return new BoostCardInfo(
         BoostCardType.UNISWAP_NFT,
         computePoolAddress(position),
         currentTick,
         position.token0,
         position.token1,
-        colors.get(index)?.[0] ?? 'red',
-        colors.get(index)?.[1] ?? 'blue',
-        position
+        DEFAULT_COLOR0,
+        DEFAULT_COLOR1,
+        position,
+        // TODO: fetch fees earned for Uniswap NFT
+        { amount0: GN.zero(position.token0.decimals), amount1: GN.zero(position.token1.decimals) },
+        null
       );
     });
-  }, [colors, nonZeroUniswapNFTPositions, slot0Data]);
+  }, [uniswapNFTPositions, slot0Data]);
+
+  /*//////////////////////////////////////////////////////////////
+                           COMPUTE COLORS
+  //////////////////////////////////////////////////////////////*/
+  useEffect(() => {
+    (async () => {
+      const merged = boostedCardInfos.concat(uniswapCardInfos);
+      const tokenAddresses = Array.from(
+        new Set(merged.flatMap((cardInfo) => [cardInfo.token0.logoURI, cardInfo.token1.logoURI])).values()
```

Comment:
Since many tokens will be repeated, I used a Set here to make sure we only compute prominent color for each token once.